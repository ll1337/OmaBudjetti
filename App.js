import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, StrictMode } from 'react-native';


export default function App() {
  return (
    <StrictMode>
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
      <StatusBar style="auto" />
    </View>
    </StrictMode>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
