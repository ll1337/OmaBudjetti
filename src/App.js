import * as React from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet } from 'react-native';
import { StrictMode } from 'react';
import store from './app/store';
import { Provider as StoreProvider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Provider as PaperProvider } from 'react-native-paper';
import Home from './pages/home.js'
import Landing from './pages/landing.js'
import Welcome from './pages/welcome'
import Living from './pages/living.js';
import Travel from './pages/travel.js';
import Loans from './pages/loans.js';
import Health from './pages/health.js';
import Insurance from './pages/insurance.js';
import OtherExpenses from './pages/otherexpenses.js';
import RandomExpenses from './pages/randomexpenses.js';
import Year from './pages/year';
import Break from './pages/break';
import Incomes from './pages/incomes';
import BottomNavBar from './components/bottomNavBar';
import EditMonth from './pages/editMonth';
import EditEvent from './pages/editEvent';

export default function App() {

  const Stack = createNativeStackNavigator();

  return (
    <StrictMode>
      <StoreProvider store={store}>
        <PaperProvider>
          <NavigationContainer>
            <Stack.Navigator
              initialRouteName="Landing"
              screenOptions={{
                headerShown: false
              }}
            >
              <Stack.Screen
                name="Home"
                component={Home}
                options={{
                  title: "Eva - OmaBudjetti",
                  headerStyle: {
                    backgroundColor: '#17B5AD',
                  },
                  headerTintColor: '#FFFFFF',
                  headerTitleAlign: 'center',
                  headerTitleStyle: {
                    fontFamily: 'Roboto',
                    fontWeight: 'bold',

                  },
                }}
                style={styles.container}
              />
              <Stack.Screen
                name="Landing"
                component={Landing}
                options={{ title: "Landing" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Welcome"
                component={Welcome}
                options={{ title: "Welcome" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Living"
                component={Living}
                options={{ title: "Living" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Travel"
                component={Travel}
                options={{ title: "Travel" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Loans"
                component={Loans}
                options={{ title: "Loans" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Health"
                component={Health}
                options={{ title: "Health" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Insurance"
                component={Insurance}
                options={{ title: "Insurance" }}
                style={styles.container}
              />
              <Stack.Screen
                name="OtherExpenses"
                component={OtherExpenses}
                options={{ title: "OtherExpenses" }}
                style={styles.container}
              />
              <Stack.Screen
                name="RandomExpenses"
                component={RandomExpenses}
                options={{ title: "RandomExpenses" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Break"
                component={Break}
                options={{ title: "Break" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Incomes"
                component={Incomes}
                options={{ title: "Incomes" }}
                style={styles.container}
              />
              <Stack.Screen
                name="Year"
                component={Year}
                options={{ title: "Year" }}
                style={styles.container}
              />
              <Stack.Screen
                name="EditEvent"
                component={EditEvent}
                options={{ title: "EditEvent" }}
                style={styles.container}
              />
              <Stack.Screen
                name="EditMonth"
                component={EditMonth}
                options={{ title: "EditMonth" }}
                style={styles.container}
              />
              <Stack.Screen
                name="BottomNavBar"
                component={BottomNavBar}
              />
            </Stack.Navigator>
            <StatusBar style="auto" />
          </NavigationContainer>
        </PaperProvider>
      </StoreProvider>
    </StrictMode>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFFFF',
    alignItems: 'center',
    justifyContent: 'center',
    fontFamily: 'Roboto',
  },
});
