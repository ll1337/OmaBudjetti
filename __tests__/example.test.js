import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

afterEach(() => {
  jest.useRealTimers();
});

beforeEach(() => {
  jest.useFakeTimers();
});

describe("example testing", () => {

  test('renders correctly', async () => {
    const tree = renderer.create(<App />).toJSON();
    await expect(tree).toMatchSnapshot();
  });
});

