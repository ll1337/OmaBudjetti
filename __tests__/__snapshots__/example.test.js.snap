// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example testing renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Landing"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      undefined,
                    ]
                  }
                >
                  landed
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
